worker/formatting_stack/kondo_classpath_cache.clj

```diff
@@ -17,7 +17,7 @@
 (def cache-dir (str cache-parent-dir File/separator cache-subdir))
 
 (def classpath-cache
-  (future
+  (delay
```

Comment:
I'd suggest a `(delay (future x))` so that one gains some responsiveness in some cases.

...Adding comments for both constructs so that we don't forget these intricacies

Reply:
Maybe I'd be a bit more comfortable with not losing the responsiveness gain, particularly given this is a problem that can be avoided by setting a generous `{:repl-options {:timeout` (which IME tends to be necessary anyway for other reasons in large projects).

FWIW I have never had the described problem

How about a macro that does: `(if (System/getProperty ...) (future ...) (delay ...))`

I would probably have no problem defaulting to slowness, but one should be able to not lose perf

Reply:
Sure thing, we're just testing this approach right now.

 Having said that; my personal lein-profile has a repl-timeout of several minutes; and I still experienced timeouts on first boot. 
 - This could be mitigated by not requiring f-s. 
 - Another pointer was that the kondo cache thread would be using a fair amount of cpu time; competing with the repl class loader.
 - retrying to load the repl would result in a booting repl at some point, once the cache was (mostly) created.

I think in hindsight that creating a cache (and thus reading analyzing file on the classpath) upon require is a rather expensive approach. 


---

About the `(delay (future ..` suggestion; i don't see any value in that, could you elaborate on the `responsiveness in some cases` ?. Consumers could still wrap the dereferencing in a future if they'd want to calculate the cache on require.